pep-9999.rst

```diff
@@ -0,0 +1,158 @@
+metadata specification for Python packages. Version 2.1 is specified in PEP
+566.
+
+The key and only change introduced in this PEP is how license is documented in
+package metadata.
+
+Software is licensed and providing accurate licensing information to Python
+packages users is an important matter.  Yet, license metadata are limited often
+leading to some confusion. 
+
+- There are two overlapping places where a license can be documented leading to
+  confusion for packagers: license-related `Classifiers` and the `License`
+  attribute.
+
+- Furthermore, it is difficult to convey anything but simple licensing using
+  these two fields. License information expressiveness and clarity when creating
+  packages will help packagers and users alike.
+
+License expressions provide a better way to express the license of a
+distribution using a well defined syntax and well known license ids from SPDX.
+
+This PEP was started based on several discussions and in particular:
+
+- `<https://github.com/pypa/warehouse/issues/2996>`_
+- `<https://github.com/pypa/interoperability-peps/issues/46>`_
+- `<https://github.com/pypa/packaging-problems/issues/41>`_
+- `<https://github.com/pypa/wheel/issues/138>`_
+- `<https://github.com/pombredanne/spdx-pypi-pep/issues/1#issuecomment-281343077>`_
+
+
+Fields
+======
+
+The canonical source for the names and semantics of each of the supported
+metadata fields is the `Core Metadata Specification`_ document.
+
+The details for the update to the `Core Metadata Specification`_ document are 
+in this `Core Metadata Specification pull request`_.
+
+
+New in Version 2.2
+------------------
+
+License-Expression (optional)
+:::::::::::::::::::::::::::::
+
+A string containing a license expression using the SPDX license expression
+syntax and SPDX-listed license identifiers so that user can express accurately
+the licensing of a package in a way that is both easy to read for humans and
+easy to process automatically by tools.
+
+A string containing a license expression using `SPDX license expression syntax and
+identifiers <https://spdx.org/ids>`__ that indicates the license(s) covering the
+distribution. The license expressions are specified in the `SPDX specification 
+<https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60>`_.
+SPDX also publishes a companion `list of license ids <https://spdx.org/license-list>`_
+for use in license expressions. 
+
+Note: this field is not case sensitive though the normalized representation of
+an SPDX license expression has a well defined case.
+
+Examples::
+
+    License-Expression: BSD-3-Clause
+
+    License-Expression: MIT OR GPL-2.0-or-later AND (FSFUL AND BSD-2-Clause)
+
+    License-Expression: GPL-3.0-or-later WITH Classpath-Exception-2.0 OR BSD-3-Clause
+
+
+Tools `should` validate that this field contains a valid license expression
+and `should` report a warning if the field contains an invalid expression.
```

Comment:
Both the SPDX license expression syntax and the license list themselves are versioned, so we should probably be clear in this section that we're incorporating them in a "this version or later compatible version" sense, not a strict "exactly this version" sense.